compose/foundation/foundation/src/uikitMain/kotlin/androidx/compose/foundation/text/input/internal/selection/TextFieldMagnifierNode.uikit.kt

```diff
@@ -0,0 +1,240 @@
+    visible: Boolean
+): TextFieldMagnifierNode {
+    return if (isPlatformMagnifierSupported()) {
+        TextFieldMagnifierNodeImpl(
+            textFieldState = textFieldState,
+            textFieldSelectionState = textFieldSelectionState,
+            textLayoutState = textLayoutState,
+            visible = visible
+        )
+    } else {
+        object : TextFieldMagnifierNode() {
+            override fun update(
+                textFieldState: TransformedTextFieldState,
+                textFieldSelectionState: TextFieldSelectionState,
+                textLayoutState: TextLayoutState,
+                visible: Boolean
+            ) {
+            }
+        }
+    }
+}
+
+internal class TextFieldMagnifierNodeImpl(
+    private var textFieldState: TransformedTextFieldState,
+    private var textFieldSelectionState: TextFieldSelectionState,
+    private var textLayoutState: TextLayoutState,
+    private var visible: Boolean
+) : TextFieldMagnifierNode(),
+    ObserverModifierNode,
+    CompositionLocalConsumerModifierNode {
+
+    private var sourceCenter by mutableStateOf(Offset.Unspecified)
+    private var color by mutableStateOf(Color.Unspecified)
+    private var density by mutableStateOf(Density(1f,1f))
+    private var magnifierSize by mutableStateOf(DpSize.Zero)
+
+    private val magnifierNode = delegate(
+        MagnifierNode(
+            sourceCenter = { sourceCenter },
+            onSizeChanged = { magnifierSize = it },
+        )
+    )
+
+    private var positioningJob : Job? = null
+
+    override fun onAttach() {
+        super.onAttach()
+        onObservedReadsChanged()
+        restartPositionJob()
+    }
+
+    override fun onObservedReadsChanged() {
+        observeReads {
+            color = currentValueOf(LocalTextSelectionColors).handleColor
+            density = currentValueOf(LocalDensity)
+
+            magnifierNode.update(color = color)
+        }
+    }
+
+    override fun update(
+        textFieldState: TransformedTextFieldState,
+        textFieldSelectionState: TextFieldSelectionState,
+        textLayoutState: TextLayoutState,
+        visible: Boolean
+    ) {
+        val previousTextFieldState = this.textFieldState
+        val previousSelectionState = this.textFieldSelectionState
+        val previousLayoutState = this.textLayoutState
+        val wasVisible = this.visible
+
+        this.textFieldState = textFieldState
+        this.textFieldSelectionState = textFieldSelectionState
+        this.textLayoutState = textLayoutState
+        this.visible = visible
+
+        if (textFieldState != previousTextFieldState ||
+            textFieldSelectionState != previousSelectionState ||
+            textLayoutState != previousLayoutState ||
+            visible != wasVisible
+        ) {
+            restartPositionJob()
+        }
+    }
+
+    private fun restartPositionJob() {
+        positioningJob?.cancel()
+        if (visible) {
+            positioningJob = coroutineScope.launch {
```

Comment:
It seems like we don't need to restart position job because we're not running animation like Android does.
Also we can simplify `update` method.